### ISSUE TITLE
Attendee Stickiness Confusing

### ISSUE THREAD

**Issue by harrislapiroff** [littleweaver/django-brambling#528]
#454 creates a confusing UI situation. Say I do the following:

1. Begin the checkout process, getting past filling in attendee details
2. Get distracted from the computer for a bit. Let the cart timer run out.
3. Go back to the shop to actually do the checkout process. Add a pass to my cart and begin the checkout process.

I will get presented with this screen:

![screenshot 2015-05-19 11 14 13](https://cloud.githubusercontent.com/assets/296631/7706389/331c8eac-fe18-11e4-9415-a61f4e828788.png)

which, A. feels bad because it looks like an error and I don't think I did anything wrong and B. doesn't make it obvious what I'm supposed to do next. (Especially since I have never done item assignment before because I wasn't required to do it the first time through, just purchasing one item for one attendee.)

**Comment by melinath** [comment]
I would argue that this is a flaw that's always been there, which is now more obviously exposed. The item assignment ui is terrible and unintuitive and always has been. :-p

**Comment by harrislapiroff** [comment]
Yeah, but if a person is just purchasing on item for one attendee, they shouldn't have to assign anything, is my feeling about it.

**Comment by harrislapiroff** [comment]
I'm not saying #454 needs to be reverted, if that's what you're thinking. I just think this needs to work more smoothly and the process should (ideally) not bring me to completely new screens because I had a cart which expired previously.

**Comment by harrislapiroff** [comment]
(I opened this ticket because I encountered this UI and was confused for a minute and I can imagine that someone who, unlike me, doesn't understand how Dancerfly works would just give up at this point because it looks like there are two errors, no fields to correct, and no "next" button to move on with.)

**Comment by melinath** [comment]
Just to be clear, I agree that this would be very confusing. I guess all I mean to say is that I'd rather fix this by making the assignment ui more intuitive rather than with more special casing. Of we could get rid of the special casing we already have, that would be even better.

**Comment by harrislapiroff** [comment]
But that seems _more_ complicated. Again, I think we should optimize for the standard case, which is someone checking out for one attendee. Forcing them to assign items when there _is_ only one attendee is extra work and an additional potential point for a user to get confused.

**Comment by harrislapiroff** [comment]
It is a UX regression from what we had before.

**Comment by harrislapiroff** [comment]
(Extra work in the sense that it is arguably unnecessary. How many ways are there of assigning one item to one attendee? Why are we providing a screen making them do that?)

**Comment by melinath** [comment]
From a user:

> I got almost to the end of registering myself (as a new, not-logged-in attendee), and then backed up to the screen where I entered my eMail address.  Then I hit save and continue.  This apparently added me as an attendee twice.  I don't see how to remove the duplicate.  Indeed the whole function of [the attendees view] wasn't clear to me at first.  Now that I think about it I can see that you are trying to create payment for multiple people, each of whom can order different goods and services.  Adding the duplicate attendee got me stuck.  Nothing to do but bail out, clear all cookies and start over.  Indeed, attempts to add a second attendee end in gridlock.  There's no way to enter item selections for the second and subsequent attendees.

**Comment by spilzer** [comment]
I ran into a similar problem, where as an organizer I was testing my event to see what adding an attendee would feel like from a user perspective and now I'm stuck with a fake attendee called "My Friend". Maybe the easiest thing to do would be to allow people to easily delete attendees?
